lib/LoginStrategy.ts

```diff
@@ -239,4 +254,78 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or null
+   */
+  async verifyClaim(token: string, payload: ITokenPayload): Promise<string> {
+    const didDocument = await this.cacheServerClient?.getDidDocument(payload.iss) ?? await this.didResolver.read(payload.iss)
+
+    if (await this.isAuthorized(token, didDocument))
+      return payload.iss
+    return null 
+  }
+  
+  private isAuthorized = async (claimedToken: string,didDocument: IDIDDocument) : Promise<boolean> => {
+    //read publickey field in DID document
+    const didPubKey = didDocument.publicKey
+    if (didPubKey.length === 0)
+      return false
+
+    const keys = new Keys({privateKey: this.privateKey})
+    const jwtSigner =  new JWT(keys)
+
+    //get all authenticated public keys
+    const authenticatedPubkeys = didDocument.publicKey.filter(pubkey => {
+      return this.isAuthenticated(pubkey, didDocument.authentication)
+    })
+
+    const validKeys = await this.filterValidKeys(authenticatedPubkeys, async (pubKeyField) => {
+      try  {
+       const decodedClaim = await jwtSigner.verify(claimedToken, pubKeyField["publicKeyHex"].split('x')[1]);
+        return decodedClaim !== undefined;
+      }
+      catch(error){
+        return false
+      }
+    })
+    return validKeys.length !== 0
+  }
+
+  private filterValidKeys = async (authenticatedKey: IPublicKey[], verifSignature) => {
+    const results = await Promise.all(authenticatedKey.map(verifSignature));
+
+    return authenticatedKey.filter((_key, index) => results[index]);
+  }
+
+  private isAuthenticated = (publicKey: IPublicKey, authFieldDocument: (string | IAuthentication)[]) => {
+    const authenticatedKeys = authFieldDocument.map(auth => {
+      if (auth.length === 0 && publicKey !== undefined)
+       return this.isSigAuth(publicKey["type"])
```

Comment:
I think this might be an issue. I think that the `isSigAuth` case is applicable even if `auth.length !== 0`

Reply:
Yes, the way I think about this is have a particular look on sigAuth type when we have the case where authentication field is empty. But if `auth.length !== 0` it will proceed with the normal bind checking using the `areLinked` function. Thus, a `sigauth` publickey type is handled in this scenario the sameway as a `verikey`.  Am I missing something ?

Reply:
This is an example of a case with two valid "authentication" keys
```
{
  "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E",
  "service": [],
  "authentication": [
    {
      "type": "owner",
      "validity": {
        "_hex": "0x1fffffffffffff"
      },
      "publicKey": "my-favorite-public-key-1"
    }
  ],
  "publicKey": [
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1veriKey",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    },
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1sigAuth",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]
}
```